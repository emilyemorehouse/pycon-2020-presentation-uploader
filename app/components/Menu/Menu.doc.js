import { describe, PropTypes } from 'react-desc'

export const doc = Menu => {
  const DocumentedMenu = describe(Menu)
    .description('A Menu')
    .usage(
      `import { Menu } from 'components/Menu';
<Menu />`,
    )

  DocumentedMenu.propTypes = {
    id: PropTypes.string
      .description('The DOM id attribute value to use for the element.')
      .defaultValue(false),
  }

  return DocumentedMenu
}
