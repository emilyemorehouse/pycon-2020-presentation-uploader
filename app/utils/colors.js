const colors = {
  yellow: 'rgb(225, 189, 71);',
  crimson: 'rgb(230,34,45)', // Crimson
  magenta: 'rgb(221,17,117)', // Magenta
  modBlue: 'rgb(65,86,166)', // Mod Blue
  cuttleBlue: 'rgb(24,192,221)', // Cuttle Blue
  darkGrey: 'rgb(68,68,68)', // Dark Grey
  mediumGrey: 'rgb(170,170,170)', // Medium Grey
  lightGrey: 'rgb(238,238,238)', // Light Grey
}

colors.primary = colors.modBlue
colors.secondary = colors.yellow
colors.primaryText = colors.darkGrey
colors.background = colors.lightGrey

export default colors
