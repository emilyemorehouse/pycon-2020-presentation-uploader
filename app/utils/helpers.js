import { isPlainObject } from 'lodash'

const ERROR_DEFAULT = 'An error occurred. Please try again.'
const ERROR_NO_INTERNET = 'Request could not be made. Please check your internet connection.'

export const createWithDoc = ({ envName = '', docFunction = () => {}, component = '' }) => {
  let createComponentWithDoc
  if (envName !== 'production') {
    createComponentWithDoc = docFunction(component) // eslint-disable-line global-require
  }
  return createComponentWithDoc || component
}

export const getErrorMessage = err => {
  // - Axios does not return a response object if the service cannot be reached
  if (!err.response) {
    return ERROR_NO_INTERNET
  }

  // - If we do have a response and data object, use the full error object
  if (err.response && err.response.data) {
    try {
      const errorStringOrObject = Object.values(err.response.data)[0]
      if (isPlainObject(errorStringOrObject)) return errorStringOrObject[0]
      return errorStringOrObject
    } catch (error) {
      // eslint-disable-next-line no-console
      console.warn('Could not infer error message from error response object')
    }
  }

  // If a message has not been set, fallback to our default message
  return ERROR_DEFAULT
}
