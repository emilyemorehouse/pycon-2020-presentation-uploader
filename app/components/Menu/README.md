<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## Menu
A Menu

## Usage

```javascript
import { Menu } from 'components/Menu';
<Menu />
```

## Properties

**id**

The DOM id attribute value to use for the element.

```
string
```
  