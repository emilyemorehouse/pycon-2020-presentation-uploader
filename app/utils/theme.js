import colors from './colors'

const theme = {
  global: {
    font: {
      family: 'Roboto Mono',
      size: '16px',
    },
    colors: {
      brand: colors.primary,
      focus: colors.secondary,
    },
  },
  button: {
    border: {
      radius: '4px',
    },
  },
}

export default theme
