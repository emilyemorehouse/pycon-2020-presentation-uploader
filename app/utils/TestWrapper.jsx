/* eslint-disable react/prop-types */
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import { Grommet } from 'grommet'

import theme from './theme'

export const TestWrapper = ({ children }) => (
  <IntlProvider locale="en">
    <MemoryRouter initialEntries={['/']}>
      <Grommet theme={theme}>{children}</Grommet>
    </MemoryRouter>
  </IntlProvider>
)
