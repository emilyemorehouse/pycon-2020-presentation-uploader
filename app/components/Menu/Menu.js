/* eslint-disable react/prop-types */
import React from 'react'

// Components
import { Menu as GrommetMenu } from 'grommet'

// Helpers
import { doc } from './Menu.doc'
import { createWithDoc } from '../../utils/helpers'

/**
 *
 * Menu
 *
 */
const Menu = ({ children, ...rest }) => <GrommetMenu {...rest}>{children}</GrommetMenu>

export default createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: Menu,
})
