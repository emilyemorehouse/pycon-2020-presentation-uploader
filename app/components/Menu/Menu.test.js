import React from 'react'
import { render } from '@testing-library/react'
// import '@testing-library/jest-dom/extend-expect' // add some helpful assertions

// Components
import { TestWrapper } from 'utils/TestWrapper'
import Menu from './Menu'

const renderComponent = (props = {}) =>
  render(
    <TestWrapper>
      <Menu {...props} />
    </TestWrapper>,
  )

/**
 *
 * Tests for Menu
 *
 */
describe('Menu', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
