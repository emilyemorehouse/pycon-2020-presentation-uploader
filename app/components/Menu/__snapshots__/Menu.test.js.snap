// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 blnwXx"
  >
    <button
      aria-label="Open Menu"
      class="StyledButton-sc-323bzc-0 dnCEuz"
      type="button"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 hYQclW"
      >
        <span
          class="StyledText-sc-1sadyjn-0 hUokoe"
        />
        <svg
          aria-label="FormDown"
          class="StyledIcon-ofa7kd-0 dTtTsP"
          viewBox="0 0 24 24"
        >
          <polyline
            fill="none"
            points="18 9 12 15 6 9"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;
